crates/subspace-networking/src/utils/tests.rs

```diff
@@ -60,3 +60,74 @@ fn test_batching() {
     assert_eq!(batcher.next_batch(collection.clone()), vec![3, 4, 5, 6]);
     assert_eq!(batcher.next_batch(collection), vec![7, 1, 2, 3]);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_resizable_semaphore_alloc() {
+    // Capacity = 3. We should be able to alloc only three permits.
+    let sem = ResizableSemaphore::new(3);
+    let (_permit_1, _permit_2, _permit_3) = (
+        sem.try_acquire().await.unwrap(),
+        sem.try_acquire().await.unwrap(),
+        sem.try_acquire().await.unwrap(),
+    );
+    assert!(sem.try_acquire().await.is_none());
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_resizable_semaphore_expand() {
+    // Initial capacity = 3.
+    let sem = ResizableSemaphore::new(3);
+    let (_permit_1, _permit_2, _permit_3) = (
+        sem.try_acquire().await.unwrap(),
+        sem.try_acquire().await.unwrap(),
+        sem.try_acquire().await.unwrap(),
+    );
+    assert!(sem.try_acquire().await.is_none());
+
+    // Increase capacity of semaphore by 2, we should be able to alloc two more permits.
+    sem.expand(2).await;
+    let (_permit_4, _permit_5) = (
+        sem.try_acquire().await.unwrap(),
+        sem.try_acquire().await.unwrap(),
+    );
+    assert!(sem.try_acquire().await.is_none());
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_resizable_semaphore_shrink() {
+    async fn free_permit(permit: ResizableSemaphorePermit) {
+        tokio::spawn({
+            async move {
+                std::mem::drop(permit);
+            }
+        })
+        .await
+        .unwrap();
+    }
```

Comment:
This is identical to:
```suggestion
    fn free_permit(_permit: ResizableSemaphorePermit) {
    }
```

Which in turn is identical to `drop(permit)`.

There is nothing blocking in `Drop` impl to require spawning a task, and even if there was, you'd want a blocking tasks in a dedicated thread. Thankfully, none of it is necessary, `free_permit(..).await;` can be just replaced with `drop(..);`.